modules/editor-service-api/src/main/java/org/opencastproject/editor/api/EditorRestEndpointBase.java

```
@@ -87,6 +91,63 @@ public Response getEditorData(@PathParam("mediaPackageId") final String mediaPac
     }
   }
 
+  @POST
+  @Path("{mediaPackageId}/lock")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "lockMediapackage",
+          description = "Creates and updates the lock for a mediapackage in the editor",
+          returnDescription = "",
+          pathParameters = {
+            @RestParameter(name = "mediaPackageId", description = "The id of the media package", isRequired = true,
+                    type = RestParameter.Type.STRING)
+          },
+          responses = {
+            @RestResponse(description = "Lock obtained", responseCode = SC_CREATED),
+            @RestResponse(description = "Lock not obtained", responseCode = SC_CONFLICT),
+            @RestResponse(description = "Mediapackage not found", responseCode = SC_NOT_FOUND)
+          })
+  public Response lockMediapackage(@PathParam("mediaPackageId") final String mediaPackageId,
+         @Context HttpServletRequest request) {
+    try {
+      String details = readInputStream(request);
```

`readInputStream`, at least when testing with the rest docs, always returns an empty string.  It's unclear what `details` is *supposed* to be, but because it's empty `lockData` is null, which causes the lock process to fail [here](https://github.com/opencast/opencast/pull/4658/files#diff-22ae4a40c4d0c69ab7f0385e1853c759f72647351f5b58866fd90eb97d3ae475R59).

---

This is only intend to complete the remote API. Locking is POST operation and to make re-use of the existing doPostForMediaPackage() function the lock object is converted to JSON and sent as the request body. 
Like the other POST's in this API I guess it should return an 400 error if no body is present.

---

I relented and changed to using form data so the REST docs make sense. I'd gone down the route of trying to use the existing request code here and in the opencast-editor which just has a simple post(json) call. That PR https://github.com/opencast/opencast-editor/pull/1000 will need additional work now too.